applications/vanilla/modules/class.discussionssortfiltermodule.php

```diff
@@ -197,8 +203,26 @@ protected function getFilterDropdowns() {
                     [],
                     ['rel' => 'nofollow']
                 );
+
+                $link = $dropdown->getItems()[val('group', $filter)]['items'][val('key', $filter)];
```

Comment:
For the sake of readability, I would highly recommend breaking this up a little bit.

1. The result of `$dropdown->getItems` can be moved outside the `foreach` and assigned to a variable. That variable can be referenced in the loop, without needing to invoke the function with every iteration.
1. You're using the value of "group" on the next line. Maybe move the assignment to the top of the loop and use its value here. Similarly, you could assign all the values you'll be using from `$filter` (or other collections) into their own variable near the top of the loop, if you'll be using them as keys or for comparisons.

```php
$link = $items[$group]['items'][$filterKey] // Looks like you're assigning $key to another value. Otherwise, I'd recommend using that.
```